src/app/main.cpp

```diff
@@ -235,19 +234,25 @@ int main(int argc, char *argv[])
 #if defined(DISABLE_GUI) && !defined(Q_OS_WIN)
         if (params.shouldDaemonize)
         {
-            app.reset(); // Destroy current application
-            if (daemon(1, 0) == 0)
+            app.reset(); // Destroy current application instance
+            if (::daemon(1, 0) == 0)
             {
                 app = std::make_unique<Application>(argc, argv);
-                if (app->isRunning())
+                if (app->hasAnotherInstance())
                 {
-                    // Another instance had time to start.
+                    const QString errorMessage = QCoreApplication::translate("Main", "Found unexpected qBittorrent instance. Exiting this instance. Current process ID: %1.")
+                        .arg(QString::number(QCoreApplication::applicationPid()));
+                    LogMsg(errorMessage);
```

Comment:
I would consider it undefined behavior to write something to log if `app->hasAnotherInstance()` is `true`.

Reply:
Now that I notice other problems.
Here the `::daemon()` succeeded and therefore stderr is closed and that means `qCritical()` won't show the message to console. `qCritical()` still has other side effects but I'm not sure they are important.
So here we are left with `LogMsg` to record the error.

>I would consider it undefined behavior to write something to log if app->hasAnotherInstance() is true.

Yes, it seems the log file could be overwritten by the other qbt instance.

Maybe it is still worth write to log because there are no other way to record the error... what do you think?
Also if this is accepted, I should add a comment about this weak effort.

Reply:
Okay, let's leave it that way.